src/game/ObjectMgr.cpp

```diff
@@ -1975,6 +1975,38 @@ PetLevelInfo const* ObjectMgr::GetPetLevelInfo(uint32 creature_id, uint32 level)
     return &itr->second[level - 1];                         // data for level 1 stored in [0] array element, ...
 }
 
+PetFamilyStatMods const* ObjectMgr::GetPetFamilyStatMods(uint32 family) const
+{
+    // Loading levels data
+    //                                                   0      1     2     3
+    QueryResult* result  = WorldDatabase.Query("SELECT family, healthModifier, armorModifier, damageModifier FROM pet_familystats");
+
+    PetFamilyStatMods* pFamilyMods;
```

Comment:
Initialization missing.

However, a better solution would be selecting all data on startup and completely storing it in memory using a suitable container (in this case std::map<uint32 /* Family */, PetFamilyStatMods*>). The lookup would be much faster, more proper and better fitting into the current design.

Awesome work by the way. Keep it up! :-)